pkgs/by-name/cm/cmake/setup-hook.sh

```diff
@@ -1,5 +1,14 @@
 addCMakeParams() {
-    addToSearchPath CMAKE_PREFIX_PATH $1
+    addToSearchPath CMAKE_INCLUDE_PATH "$1/include"
+    addToSearchPath CMAKE_INCLUDE_PATH "$1"
```

Comment:
because this replicates what `CMAKE_PREFIX_PATH` does. it adds `include` to the search path first and then the bare directory last. there is code which relies on this working. looking at the edit in https://github.com/NixOS/nixpkgs/pull/318226#issuecomment-2175370938 it seems like plasma5Packages.libksysguard  has some custom module finder that requires the bare directory to find headers because it adds `include` suffix to the search path...

https://github.com/KDE/libksysguard/blob/b1b4989cf630122f3d58868a884e624a620eaf03/cmake/FindNL.cmake#L36-L38

Reply:
I think the way `PATH_SUFFIXES` work is they are checked in addition to the prefix, but maybe I'm wrong